examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -41,101 +50,346 @@ def freq_mask(freqs, fmin, fmax):
-                method=self.estimator)
+            # subtract signal:
+            inst_noise._data[self.picks_] -= inst_signal._data[self.picks_]
+           
+            cov_signal = compute_raw_covariance(
+                inst_signal, picks=self.picks_, method=self.estimator, rank=self.rank)
+            cov_noise = compute_raw_covariance(inst_noise, picks=self.picks_,
+                method=self.estimator, rank=self.rank)
             del inst_noise
             del inst_signal
         else:
-            raise NotImplementedError()
+            if isinstance(inst, BaseEpochs)  or isinstance(inst, np.ndarray):
+            # data X is epoched 
+            # part of the following code is copied from mne csp
+                n_epochs, n_channels, n_samples = inst.shape
+                self.max_components=n_channels
+              
+                #reshape for filtering
+                X_aux=np.reshape(inst, [n_epochs, n_channels*n_samples])
+                X_s=filter_data(X_aux, self.sampling_freq, **self.filt_params_signal)  
+                
+                #rephase for filtering
+                X_aux=np.reshape(inst, [n_epochs, n_channels*n_samples])
+                X_n=filter_data(X_aux, self.sampling_freq, **self.filt_params_noise)
+                # subtract signal:
+                X_n -= X_s
+                
+                # Estimate single trial covariance
+                #reshape to original shape
+                X_s=np.reshape(X_s, [n_epochs,n_channels,n_samples])
+                self.Xs=X_s
+                covs = np.empty((n_epochs, n_channels, n_channels))
+                for ii, epoch in enumerate(X_s):
+                    covs[ii] = _regularized_covariance(
+                        epoch, reg=self.estimator, method_params=self.cov_method_params,
+                        rank=self.rank)
+                    
+                cov_signal = covs.mean(0)
+            
+                #% Covariance matrix for the flanking frequencies (noise)
+                        
+                
+                
+                #reshape to original shape
+                X_n=np.reshape(X_n, [n_epochs,n_channels,n_samples])
+                # Estimate single trial covariance
+                covs_n = np.empty((n_epochs, n_channels, n_channels))
+                for ii, epoch in enumerate(X_n):
+                    covs_n[ii] = _regularized_covariance(
+                        epoch, reg=self.estimator, method_params=self.cov_method_params,
+                        rank=self.rank)
+        
+                cov_noise = covs_n.mean(0)
+                       
+            
+            else:
+                raise NotImplementedError()
+        
+        eigvals_,eigvects_ = eigh(cov_signal.data, cov_noise.data)
+        # #sort in descencing order
+        ix = np.argsort(eigvals_)[::-1]
+        self.eigvals_=eigvals_[ix]
+        self.filters_ = eigvects_[:,ix]
+        # self.filters_ = eigvects_
```

Comment:
please remove commented out code. Documentation-related comments are ok. For code: Either we used it or it does not exist.

Reply:
done !